### PR TITLE
Homework 17

### PR DESCRIPTION
**Pull request by Bogin404**
please help me find better selectors

### CODE REVIEW

**Comment by duykhanhrc** [comment]
Hi @Bogin404,
I have checked your homework 17, test script passed. You're doing great! Keep up the good work!
Best wishes from the HW team (Khanh, khanh@testpro.io)

image

**Comment by duykhanhrc** [comment]
By the way, about your question `please, help me find better selectors - had trouble with that`
I have some suggestions as follows:
The best and most simple selector will always be to use an element ID: #some-id-here. If only we were so lucky to have this every time.
When writing selectors, you should prioritize finding in this order:
1. ID, name, class, or anything else that is unique to the element
2. Complex CSS selectors
3. XPath selectors
4. If none of the above work for you, then the last resort is to ask a developer to add a unique ID or class to the element you

File: src/test/java/BaseTest.java [hunk 1]
```diff
@@ -1,11 +1,135 @@
...
+    //
+
+    protected void verifySuccessMessageText() {
+        WebElement successNewPlaylist = driver.findElement(By.xpath(
+                "//div[contains(text(), 'new playlist17')]"));
+        Assert.assertTrue(successNewPlaylist.isDisplayed());
+    }
+
+    protected void selectSavePlaylist() {
+        WebElement savePlaylistButton = driver.findElement(By.xpath("//*[@id='songResultsWrapper']/header/div[3]/div/section[2]/form/button"));
```
Comment (duykhanhrc) [inline src/test/java/BaseTest.java:104]: at this selector savePlaylistButton, it's too long. You can find the selector based on the specific attributes of the element (as type, title, ect)
//HINT:
`//*[@id='songResultsWrapper']//button[@title='Save']`

File: src/test/java/BaseTest.java [hunk 2]
```diff
@@ -1,11 +1,135 @@
...
+        savePlaylistButton.click();
+    }
+
+    protected void enterNewPlaylistName(String newPlaylistName) {
+        WebElement createPlaylistField = driver.findElement(By.xpath("//*[@id='songResultsWrapper']/header/div[3]/div/section[2]/form/input"));
```
Comment (duykhanhrc) [inline src/test/java/BaseTest.java:109]: the same here, 
//HINT: `//*[@id='songResultsWrapper']//input[@data-test='new-playlist-name']`

File: src/test/java/BaseTest.java [hunk 3]
```diff
@@ -1,11 +1,135 @@
...
+        createPlaylistField.sendKeys(newPlaylistName);
+    }
+
+    protected void clickAddTo() {
+        WebElement addToButton = driver.findElement(By.cssSelector("[class='btn-add-to']"));
+        addToButton.click();
+    }
+
+    protected void clickFirstSong() {
+        WebElement firstSong = driver.findElement(By.xpath("//*[@id='songResultsWrapper']/div/div/div[1]/table/tr"));
```
Comment (duykhanhrc) [inline src/test/java/BaseTest.java:120]: the same here, first song WebElement can define as (//*`[@id='songResultsWrapper']//tr[contains(@class,'song-item')])[1]`

File: src/test/java/BaseTest.java (hunk 1 above)
Comment (Bogin404) [reply src/test/java/BaseTest.java:104]: @duykhanhrc I appreciate the feedback!

File: src/test/java/BaseTest.java (hunk 2 above)
Comment (Bogin404) [reply src/test/java/BaseTest.java:109]: @duykhanhrc, i was also not able to use this hint, unfortunately. 
And just like with other selectors i generated, i was at first able to come up with 'close enough' selectors that would find 2 or 3 results ( that were identical lines of code), only one of which would actually select the element i needed, so it is very confusing

**Pull request merged**